Tensile/KernelWriterSource.py

```diff
@@ -1046,7 +1046,11 @@ def allocateResources(self, kernel):
     # allocate local memory
     kStr += self.endLine
     kStr += "  /* allocate local memory */" + self.endLine
-    kStr += "  %sDATA_TYPE localMemory[LDS_NUM_ELEMENTS];%s" \
+    if kernel["LocalSplitU"] > 1:
```

Comment:
Even though we add if statement here, the LDS size calculation is still inaccurate for LocalSplitU > 1 case.

In SolutionStruct.py, ldsNumElements is calculated in different ways and use the max of these values.

    # lds size is the greater of the two
    ldsNumElements = max(ldsNumElementsAB, ldsNumElementsReduction, ldsNumElementsOccupancy)
     ...
      ldsNumElements = max(ldsNumElements, ldsNumElementsRemapC)

    state["LdsNumElements"] = ldsNumElements

computeDataType is not always necessary even if LocalSplitU > 1.
Would you please use computeDataType only for necessary scenario (which should be ldsNumElementsReduction)?

Reply:
In case of ldsNumElementsRemapC , we do the following adjustment in case size of DestDataType is different from size of DataType
===
      if state["_GlobalAccumulation"]:
        computeBytes = state["ProblemType"]["ComputeDataType"].numBytes()
        multiplier = computeBytes // state["ProblemType"]["DataType"].numBytes()
      elif state["ProblemType"]["DestDataType"].numBytes() > state["ProblemType"]["DataType"].numBytes():
        # Determine ratio of output to input element size.
        # SRVW remaps output so we need to scale up resources.
        multiplier = state["ProblemType"]["DestDataType"].numBytes() // state["ProblemType"]["DataType"].numBytes()
      else:
        multiplier = 1

      ldsNumElementsRemapC *= multiplier
===

We should need similar logic for ldsNumElementsReduction (just need to multiply sizeof(computeDataType) / sizeof(DataType) as I suggested).

Reply:
@nakajee You are right. Instead of changing localsplitU reduction part, we may replace the ldsNumElementsReduction =  state["LocalSplitU"]*state["MacroTile0"]*state["MacroTile1"]
with 
ldsNumElementsReduction =  sizeof(COMPUTE_DATA_TYPE)/sizeof(DATA_TYPE)*state["LocalSplitU"]*state["MacroTile0"]*state["MacroTile1"]
and define localMemory as
__shared__ DATA_TYPE localMemory[LdsNumElements];

Reply:
@nakajee We don't need an additional if statement to deal with LocalSplitU >1 and LocalSplitU == 1

Reply:
local@nakajee There are three changes:
1. ldsNumElementsReduction should multiply by sizeof(COMPUTE_DATA_TYPE)/sizeof(DATA_TYPE)
2. localMemory is defined as type DATA_DATA which is the same as the original implementaton.
3. localMemory is typed cast to COMPUTE_DATA_TYPE when it is used to store intermediate results.